realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1092,20 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Condition that finds a distinct set of objects of a specific class. As a Realm is unordered,
+     * it is undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return the query object.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmQuery<E> distinct(String fieldName) {
+        long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.STRING);
+        this.query.distinct(columnIndices);
+        return this;
```

Comment:
How about below API?

RealmQuery:
//equivalent of findAllSorted(String fieldName)
`findAllDistinct(String fieldName)`
//equivalent of findAllSortedAsync(String fieldName)
`findAllDistinctAsync(String fieldName)`

RealResults:
//equivalent of sort(String fieldName)
`distinct(String fieldName)`

Realm:
//equivalent of allObjectsSorted(Class<E> clazz, String fieldName)
//requires renaming of already existing distinct() method
`allObjectsDistinct(Class<E> clazz, String fieldName)`

//requires renaming of already existing distincAsync() method
`allObjectsDistinctAsync(Class<E> clazz, String fieldName)`

As a side effect below sort method should also be added:
`allObjectsSortedAsync(Class<E> clazz, String fieldName)`